sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -603,7 +603,13 @@ object DateTimeUtils {
    */
   private[this] def getYearAndDayInYear(daysSince1970: SQLDate): (Int, Int) = {
     // add the difference (in days) between 1.1.1970 and the artificial year 0 (-17999)
-    val daysNormalized = daysSince1970 + toYearZero
+    var  daysSince1970Tmp = daysSince1970
+    // In history,the period(1582-10-05 ~ 1582-10-14) is not exist
```

Comment:
`In history,the period(5.10.1582 ~ 14.10.1582) is not exist`
->
`Since Julian calendar was replaced with the Gregorian calendar, the 10 days after Oct. 4 were skipped.`